src/main/interface/cli.c

```diff
@@ -512,28 +530,68 @@ static void cliPrintVarRange(const clivalue_t *var)
     case (MODE_ARRAY): {
         cliPrintLinef("Array length: %d", var->config.array.length);
     }
-
+    break;
+    case (MODE_BITSET): {
+        cliPrintLinef("Allowed values: OFF, ON");
+    }
     break;
     }
 }
 
 static void cliSetVar(const clivalue_t *var, const int16_t value)
 {
     void *ptr = cliGetValuePointer(var);
+    uint32_t workValue;
+    uint32_t mask;
 
-    switch (var->type & VALUE_TYPE_MASK) {
-    case VAR_UINT8:
-        *(uint8_t *)ptr = value;
-        break;
+    if ((var->type & VALUE_MODE_MASK) == MODE_BITSET) {
```

Comment:
This is a bit overcomplicated. Reading 32bit value and letting write code to strip it should be enough:
```
if ((var->type & VALUE_MODE_MASK) == MODE_BITSET) {
   memcpy(&workValue, ptr, sizeof(workValue)); // little endian assumed, cliGetVar may be implemented instead
   workValue = workValue & ~(1 << var->config.bitpos) | (value <<  var->config.bitpos);  // replace bit
} else {
   workValue = value;
}
switch (var->type & VALUE_TYPE_MASK) {
 case VAR_UINT8:
            *(uint8_t *)ptr = workValue;
            break;
 ...